aperturedb/DataHelper.py

```diff
@@ -0,0 +1,159 @@
+
+        files = kaggle.dataset_list_files(dataset_ref)
+        x = kaggle.dataset_download_files(
+            dataset=dataset_ref,
+            path=workdir,
+            unzip=False)
+
+        archive = None
+        for _, subdirs, dfiles in os.walk(workdir):
+            if len(dfiles) == 1 and len(subdirs) == 0:
+                archive = os.path.join(workdir, dfiles[0])
+
+                with zipfile.ZipFile(archive, 'r') as zip_ref:
+                    zip_ref.extractall(workdir)
+
+                break
+
+        if len(files.files) > 0 and indexer is None:
+            # This can be inferred from the files of the DS. Uusally is a csv, json, sqlite, etc.
+            csvs = filter(lambda f: f.fileType == ".csv", files.files)
+            index = sorted(csvs, key=lambda f: -f.totalBytes)[0]
+
+            if index is None:
+                print(
+                    f"Could not determine the index.  Files in the dataset = {files}")
+
+            print(f"Found index {index}")
+
+            collection = pd.read_csv(os.path.join(workdir, index.name))
+        else:
+            # There happen to be image files, along with some annotation files.
+            try:
+                collection = indexer(workdir)
+            except Exception as e:
+                help = f"Failed getting an index to the dataset. Specify custom indexer (https://...) ."
+                raise Exception({
+                    "exception": e,
+                    "help": help
+                })
+
+        con = Connector.Connector(user="admin", password="admin")
+        loader = ImageLoader(con)
+        ploader = ParallelLoader(con)
+
+        def process_row(row):
+            custom_row = generator(
+                dataset_ref,
+                workdir,
+                collection.columns,
+                row)
+
+            # Add some standard properties to specifically address this dataset.
+            properties = custom_row["properties"]
+            properties["index"] = row[0]
+            properties["dataset_name"] = dataset_ref
+            properties["dataset_source"] = "kaggle"
+            return custom_row
+
+        if query_generator is not None:
+            try:
+                ploader.ingest(
+                    generator=list(map(lambda row: query_generator(
+                        dataset_ref, workdir, collection.columns, row), collection.iterrows())),
+                    numthreads=1,
+                    batchsize=1,
+                    stats=True
+                )
+            except Exception as e:
+                help = f"Failed emitting a transcation for {collection.columns}."
+                raise Exception({
+                    "exception": e,
+                    "help": help
+                })
+        else:
+            try:
+                loader.ingest(
+                    generator=list(
+                        map(lambda row: process_row(row), collection.iterrows())),
+                    numthreads=1,
```

Comment:
why 1 thread and batches of 1? 
we should probably be doing this outside of this class.